src/Microsoft.Health.Dicom.Functions.UnitTests/DurableFunctionsTimedCleanupTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+//using System;
+//using System.Collections.Generic;
+//using System.Text;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Functions.UnitTests
+{
+    public class DurableFunctionsTimedCleanupTests
+    {
+        private readonly ReindexErrorConfiguration _cleanupConfig;
+        private readonly TimerInfo _timer;
+        private readonly ILogger _logger;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly DateTime _definedNow;
+        private readonly DurableFunctionsTimedCleanup _durableFunctionsCleanup;
+        private readonly IDurableOrchestrationClient _durableOrchestrationClientMock;
+
+        public DurableFunctionsTimedCleanupTests()
+        {
+            _cleanupConfig = new ReindexErrorConfiguration();
+            _cleanupConfig.RuntimeStatuses = new OrchestrationRuntimeStatus[] { OrchestrationRuntimeStatus.Completed };
+            _timer = Substitute.For<TimerInfo>(default, default, default);
+            _logger = Substitute.For<ILogger>();
+            _cancellationTokenSource = new CancellationTokenSource();
+            _definedNow = DateTime.UtcNow;
+            _durableFunctionsCleanup = new DurableFunctionsTimedCleanup(
+                Options.Create(_cleanupConfig),
+                () => _definedNow);
+            _durableOrchestrationClientMock = Substitute.For<IDurableOrchestrationClient>();
+        }
+
+        [Fact]
+        public async Task GivenValidInput_CleanupCompletedDurableFunctions__ShouldPurgeListOfOrchestrationsAsync()
```

Comment:
These methods are quite validating, as they simply run and don't check to see whether the methods were called as expected. So you'll need to leverage the `Received(int)` method in NSubstitute.

To that end, I think you'll find it easier to simply make a helper function for the `OrchestrationStatusQueryCondition` predicate, like:

```csharp
private bool IsCondition(OrchestrationStatusQueryCondition actual, string token)
{
    return actual.RuntimeStatus.SequenceEqual(_cleanupConfig.RuntimeStatuses)
        && actual.CreatedTimeFrom == DateTime.MinValue
        && actual.CreatedTimeTo == _definedNow.AddDays(-_cleanupConfig.MinimumAgeDays)
        && actual.Token == token;
}
```

Then you can say things like:

```csharp
_durableOrchestrationClientMock
    .ListInstancesAsync(
        Arg.Is<OrchestrationStatusQueryCondition>(x => IsCondition(x, null)),
        Arg.Is(tokenSource.Token))
    .Returns(Task.FromResult(...));

await _durableFunctionsCleanup.Run(...);

_durableOrchestrationClientMock
    .Received(1)
    .ListInstancesAsync(
        Arg.Is<OrchestrationStatusQueryCondition>(x => IsCondition(x, null)),
        Arg.Is(tokenSource.Token));
```